azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -281,81 +277,42 @@ def _init_impl(self):
         pass
 
     def _get_token_impl(self) -> dict:
-        # try and obtain the refresh token from AzCli
-        refresh_token = None
-        token = None
-        stored_token = self._get_azure_cli_auth_token()
-        if TokenConstants.AZ_REFRESH_TOKEN in stored_token and TokenConstants.AZ_CLIENT_ID in stored_token and TokenConstants.AZ_AUTHORITY in stored_token:
-            refresh_token = stored_token[TokenConstants.AZ_REFRESH_TOKEN]
-            self._client_id = stored_token[TokenConstants.AZ_CLIENT_ID]
-            self._authority_uri = stored_token[TokenConstants.AZ_AUTHORITY]
-            self._username = stored_token[TokenConstants.AZ_USER_ID]
-        else:
-            raise KustoClientError("Unable to obtain a refresh token from Az-Cli. Calling 'az login' may fix this issue.")
-
-        if self._msal_client is None:
-            self._msal_client = PublicClientApplication(client_id=self._client_id, authority=self._authority_uri)
-
         try:
-            token = self._msal_client.acquire_token_by_refresh_token(refresh_token, self._scopes)
-        except Exception as ex:
-            raise KustoClientError("Unable to obtain with Az-Cli refresh token. Calling 'az login' may fix this issue.\n" + str(ex))
-
-        return self._valid_token_or_throw(token, "Calling 'az login' may fix this issue.")
-
-    def _get_token_from_cache_impl(self) -> dict:
-        token = None
-        if self._msal_client is not None:
-            account = None
-            if self._username is not None:
-                accounts = self._msal_client.get_accounts(self._username)
-                if len(accounts) > 0:
-                    account = accounts[0]
-
-            token = self._msal_client.acquire_token_silent(scopes=self._scopes, account=account, client_id=self._client_id, authority=self._authority_uri)
-
-        return self._valid_token_or_none(token)
+            if self._az_auth_context is None:
+                self._az_auth_context = AzureCliCredential()
 
-    @staticmethod
-    def _get_azure_cli_auth_token() -> dict:
-        """
-        Try to get the az cli authenticated token
-        :return: refresh token
-        """
-        import os
+            self._az_token = self._az_auth_context.get_token(self._kusto_uri)

```

Comment:
Do you know if it requires azure-cli package?

Reply:
This requires az cli to be installed, as it runs az-cli as a program.